datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -2462,6 +2470,13 @@ private void registerGauge(String metricName, Supplier<?> valueSupplier) {
       _metricInfos.add(new BrooklinGaugeInfo(_coordinator.buildMetricName(MODULE, metricName)));
     }
 
+    // registers a new gauge or updates the supplier for the gauge if it already exists
```

Comment:
I added this new method since I wanted to report the change in the number of violations per datastream, hence needed a gauge for dynamic keys (datastream names). 

The register gauge registers a new gauge for a new key but for any existing key, it returns the already registered gauge and hence this function will update the supplier function for an already registered gauge metric.